gnocchi/incoming/file.py

```diff
@@ -179,3 +182,30 @@ def process_measure_for_metrics(self, metric_ids):
 
         for metric_id, files in six.iteritems(processed_files):
             self._delete_measures_files_for_metric(metric_id, files)
+
+    @contextlib.contextmanager
+    def process_measures_for_sack(self, sack):
+        measures = {}
+        processed_files = {}
+        for metric_id in self._list_target(self._sack_path(sack)):
+            try:
+                metric_id = uuid.UUID(metric_id)
+            except ValueError:
+                LOG.error("Unable to parse %s as an UUID, ignoring metric",
+                          metric_id)
+                continue
+            files = self._list_measures_container_for_metric_str(
```

Comment:
this method seems to ask for metric_id as a string but we give it as a uuid? shouldn't this break?

Reply:
No that's good, it calls str(metric_id). I admit those methods are a bit fuzzy now, I might rework/rename them in a next patch.